File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        lo, add = 0, 0
        result = int(1e9)

        for i in range(len(nums)):
            add += nums[i]
            while add >= target:
                result = min(result, i - lo + 1)
                add -= nums[lo]
                lo += 1

        return 0 if result == int(1e9) else result

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int lo = 0, add = 0, result = Integer.MAX_VALUE;
        
        for (int i = 0; i < nums.length; i++) {
            add += nums[i];
            while (add >= target) {
                result = Math.min(result, i - lo + 1);
                add -= nums[lo];
                lo++;
            }
        }

        if (result == Integer.MAX_VALUE) {
            return 0;
        }

        return result;
    }
}